qark/qarkMain.py

```diff
@@ -96,7 +96,9 @@ def apktool(pathToAPK):
     # Create /temp/foo/apktool
     # Run java -jar apktool_2.1.0.jar d /foo/bar/temp/myapp.apk --no-src --force -m --output /foo/bar/temp/apktool/
     # read AndroidManifest.xml and return the content
-    apktool = subprocess.call(['java', '-Djava.awt.headless=true','-jar', common.rootDir + '/lib/apktool_2.1.0.jar', 'd', pathToAPK, '--no-src', '--force', '-m','--output', str(pathToAPK.rsplit(".",1)[0]).rsplit("/",1)[0] + "/apktool"], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    apktool = subprocess.Popen(['java', '-Djava.awt.headless=true', '-jar', common.rootDir + '/lib/apktool_2.1.0.jar', 'd', pathToAPK, '--no-src', '--force', '-m','--output',
+        str(pathToAPK.rsplit(".",1)[0]).rsplit("/",1)[0] + "/apktool"], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    stdout, stderr = apktool.communicate()
```

Comment:
Convert this to a Popen to handle very large manifests. With the "subprocess.call", the program hangs when it encounters a manifest that is too large (4000 lines in PrebuiltGMsCore, for instance).

Reply:
Interesting, it seems that `subprocess.call` is a wrapper around `Popen` and `Popen.wait`. In the documentation for `Popen.wait` it states that if the output is too large it will cause a deadlock, I am guessing that is what happening.

https://docs.python.org/2/library/subprocess.html#subprocess.Popen.wait

Reply:
That was my assumption as well. This solution was suggested somewhere on StackOverflow (I lost the link), and fixed the problem as expected if this was the true root-cause.